src/Context/products-context.js

```diff
@@ -0,0 +1,368 @@
+  fetch("/api/user/history", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: encodedToken,
+    },
+    body: JSON.stringify({ video: passedPayload }),
+  })
+    .then((res) => res.json())
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+};
+
+const removeFromHistoryDeleteRequest = (passedPayload) => {
+  fetch(`/api/user/history/${passedPayload._id}`, {
+    method: "DELETE",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: encodedToken,
+    },
+  })
+    .then((res) => res.json())
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+};
+
+const clearHistoryDeleteRequest = () => {
+  fetch("/api/user/history/all", {
+    method: "DELETE",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: encodedToken,
+    },
+  })
+    .then((res) => res.json())
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+};
+
+const addPlaylistPostRequest = (passedPayload) => {
+  fetch("/api/user/playlists", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: encodedToken,
+    },
+    body: JSON.stringify({
+      playlist: passedPayload.mockBeePayload,
+    }),
+  })
+    .then((res) => res.json())
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+};
+
+const removePlaylistDeleteRequest = (passedPayload) => {
+  fetch(`/api/user/playlists/${passedPayload}`, {
+    method: "DELETE",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: encodedToken,
+    },
+  })
+    .then((res) => res.json())
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+};
+
+const addToPlaylistPostRequest = (passedPayload) => {
+  fetch(`/api/user/playlists/${passedPayload.playlistId}`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: encodedToken,
+    },
+    body: JSON.stringify({ video: passedPayload.video }),
+  })
+    .then((res) => res.json())
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+};
+
+const removeFromPlaylistDeleteRequest = (passedPayload) => {
+  fetch(
+    `/api/user/playlists/${passedPayload.playlistId}/${passedPayload.video._id}`,
+    {
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+        authorization: encodedToken,
+      },
+    }
+  )
+    .then((res) => res.json())
+    .then((data) => console.log(data))
+    .catch((err) => console.log(err));
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "Add to Liked":
```

Comment:
Prefer using upper case while naming action types i.e. 'ADD_TO_LIKED'

Reply:
@tejas-muthya37 I would encourage you to make constants and then use constants that is even better